test/Transition-test.js

```diff
@@ -438,5 +438,48 @@ describe('Transition', () => {
 
       wrapper.setState({ in: false })
     })
+
+
+    it('should unmount at once if not have enter timeout', done => {
+      const wrapper = mount(
+        <UnmountTransition
+          initialIn={false}
+          timeout={{ exit: 10 }}
+          onEnter={() => {
+            expect(wrapper.getStatus()).toEqual(EXITED)
+            expect(ReactDOM.findDOMNode(wrapper)).toExist()
+
+            done()
+          }}
+        />
+      ).instance()
+
+      expect(wrapper.getStatus()).toEqual(UNMOUNTED)
+      expect(ReactDOM.findDOMNode(wrapper)).toBeNull()
+
+      wrapper.setState({ in: true })
+    })
+
+
+    it('should unmount at once if not have exit timeout', done => {
+      const wrapper = mount(
+        <UnmountTransition
+          initialIn
+          timeout={{ enter: 10 }}
+          onExited={() => {
+            setTimeout(() => {
+              expect(wrapper.getStatus()).toEqual(UNMOUNTED)
+              expect(ReactDOM.findDOMNode(wrapper)).not.toExist()
+              done()
+            })
+          }}
+        />
+      ).instance()
+
+      expect(wrapper.getStatus()).toEqual(ENTERED)
+      expect(ReactDOM.findDOMNode(wrapper)).toExist()
+
+      wrapper.setState({ in: false })
+    })
```

Comment:
Simplify tests to measure that in the absence of `enter` and `exit` the component immediately entered and exited respectively.

Reply:
@silvenon i update test, but i not sure i write it right.